### PR TITLE
[cron tests] Add tool to diff results of nodejs query runs

### PR DESCRIPTION
**Pull request by chejennifer**
script to generate and output diffs between results of two nodejs query runs (run as a cronjob on gke)

TODO for future, set this differ up to run automatically and alert if there is a diff

### CODE REVIEW

File: tools/nl/nodejs_query_differ/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+This is a commmand line tool to get the diffs between two different runs of
+the Nodejs Query tests (that are run by cronjobs on gke) and outputs the diffs
+in a json that is saved locally.
+
+To run this:
+
+```
+./run.sh -e <ENV> -b <BASE_FOLDER> -t <TEST_FOLDER>
+```
+
```
Comment (shifucun) [inline tools/nl/nodejs_query_differ/README.md:12]: Add a todo to have a UI for showing the diff result?

File: tools/nl/nodejs_query_differ/differ.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  with open(output_file, 'w') as f:
+    f.write(json.dumps(report))
+  print("Done running differ")
+  print(f'Diff report saved locally to: {_OUTPUT_FILE}')
+
+
+def main(_):
+  sc = storage.Client()
+  bucket = sc.get_bucket(f'{_GCS_BUCKET}')
+  base_folder, test_folder = _get_folder_names(bucket)
```
Comment (shifucun) [inline tools/nl/nodejs_query_differ/differ.py:96]: returned folder could be empty, check that?

**Review by shifucun — APPROVED** [review]
this is very nice and clean!

File: tools/nl/nodejs_query_differ/README.md (hunk 1 above)
Comment (chejennifer) [reply tools/nl/nodejs_query_differ/README.md:12]: done

File: tools/nl/nodejs_query_differ/differ.py (hunk 2 above)
Comment (chejennifer) [reply tools/nl/nodejs_query_differ/differ.py:97]: done

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**